### ISSUE TITLE
New release ?

### ISSUE THREAD

**Issue by Guevara-chan** [boo-lang/boo#183]
I guess, there was some progress done from 2013 here and there.
How about new version distributive available without recompiling ?

Right now I'm building open-source project in Boo, so readily available compiler binaries would be nice.

**Comment by masonwheeler** [comment]
That's a good idea.

@bamboo, do we have any sort of process for producing a release?

**Comment by Guevara-chan** [comment]
Not this year, I guess ?

**Comment by eatdrinksleepcode** [comment]
If we don't have a process for producing a release, maybe we can talk about what it would take to create such a process?

**Comment by masonwheeler** [comment]
I've got a vague idea of what that would take, but I'm hoping to hear from @bamboo about this, as it's his project and he's done this before.

**Comment by bamboo** [comment]
I'm sorry but I don't have the bandwidth to help right now, do whatever you think it's the best moving forward.

**Comment by masonwheeler** [comment]
All right, I'll look into setting up a new release over the weekend.  There have been enough improvements that I agree with the OP that a new version is warranted.

**Comment by eatdrinksleepcode** [comment]
Let me know if there is anything I can do to help with this.

**Comment by masonwheeler** [comment]
@eatdrinksleepcode  Not sure.  How familiar are you with `System.Net.WebClient`?

Here's what I've got so far.  Releases are supposed to be kept up-to-date automatically by the Travis CI server.  If you look at [the build server log](https://travis-ci.org/boo-lang/boo), at the very bottom you'll see a line that invokes the `scripts/github-release.boo` script.  But there are a few problems with it.

First, if you expand it to see the build output, and you'll see an exception message: 

> System.Net.WebException: The remote server returned an error: (403) Forbidden.

According to [this SO question](https://stackoverflow.com/questions/39907742/github-api-is-responding-with-a-403-when-using-requests-request-function), this is probably due to the way the `WebClient` is set up, and can be resolved by just editing `github-release.boo` to add a user agent.

Second, it's trying to upload to the wrong repo.  Again, this can be fixed easily by editing the script.

Third, if you download [the latest release](https://github.com/boo-lang/boo/releases/tag/alpha) that did work via this method, it's not a valid ZIP file.  Looking at it with a hex editor reveals the problem: there's a bunch of HTTP message stuff stuck on the start and end of the file.  Not sure how to handle that one yet.  I think the problem is in the way `System.Net.WebClient` is uploading it, but I'm not sure yet.  Working on that now.

I'll keep this issue updated as I continue making progress.

**Comment by masonwheeler** [comment]
After updating the script, it now errors out with `401 Unauthorized` instead.  This most likely means that the OAuth token in the .travis.yml file is no longer good, which I'm just fine with.  I was honestly a bit uncomfortable seeing authorization credentials checked in to the repo in full view of the public.  I'm going to look a bit deeper into how to make this work.

**Comment by masonwheeler** [comment]
OK, making progress here.

Turns out the token isn't such a big deal, as it's encrypted with the repo's public key, and the private key is presumably only on the Travis server.  I generated a new token and got it to move to a different error state that I was prepared for, by telling it to update a release that doesn't exist.

Now to set up some releases and make this start working again!

**Comment by masonwheeler** [comment]
Getting further now, and then I hit a new error.  Time for a bit more debug code...

**Comment by masonwheeler** [comment]
Ugh.  It's giving an error that doesn't appear to be documented anywhere.  I'm going to have to take this to GitHub support and see what's going on.

**Comment by masonwheeler** [comment]
Actually I found the problem, and now I've got working uploads.  But it's still got HTTP garbage in the upload.  Working on solving that now.

**Comment by masonwheeler** [comment]
Fixed that!  We now have a working upload script that will drop the most recent build results from Travis on the [Unstable](https://github.com/boo-lang/boo/releases/tag/unstable) release.

I'll build a stable release with a new version number tomorrow.  Right now, it's late and I've got to get some sleep.  But we now have a working deploy script again. :D

**Comment by Guevara-chan** [comment]
Tomorrow. It dies.

**Comment by aaronpowell** [comment]
I'm the owner of two of the NuGet packages, https://www.nuget.org/packages/Boo and https://www.nuget.org/packages/Boo-Compiler/, I don't actually use Boo at the moment so I'm happy to transfer ownership to anyone who wants it (also, they are massively out of date!)

**Comment by masonwheeler** [comment]
@aaronpowell Yeah, I seem to be the most active maintainer at the moment.  What would the process of transferring ownership to me be like? (Asking because I haven't actually produced a NuGet package before.)
